src/s390_ecc.c

```diff
@@ -63,6 +63,19 @@ int scalar_mulx_cpacf(unsigned char *res_u,
 		      const unsigned char *u,
 		      int curve_nid);
 
+/**
+ * Since kernel 4.10 the zcrypt device driver has multi domain support and
+ * accepts CPRBs via the ioctl ZSECSENDCPRB with domain addressing 0xFFFF
+ * (AUTOSELECT_DOM in zcrypyt.h). This allows for load balancing between
+ * multiple available crypto cards.
+ */
+typedef enum {
+	dom_addressing_not_set = 0,
+	dom_addressing_autoselect,
+	dom_addressing_default_domain,
+} dom_addressing_t;
+int dom_addressing = dom_addressing_not_set;
```

Comment:
Why do we use `dom_addressing_not_set` as default here?  If I understand the remaining code correctly, on all paths where we evaluate `dom_addressing`, we first check if it is equal to `dom_addressing_not_set` and, if so, change it to `dom_addressing_autoselect`.  Couldn't we simply initialize the variable already to `dom_addressing_autoselect` and get rid of the `if` statements to change the value if not already set?  Then, we could also get rid of `dom_addressing_not_set` since it is not used anywhere anymore.  The sequence seems to be "if nothing set, set and try autoselect, if that fails set and try default domain".

Reply:
Yes, indeed. This saves some code. I made the change and re-tested on both new and old kernel.